domain/fhir/fhir_repository.go

```diff
@@ -0,0 +1,103 @@
+package fhir
+
+import (
+	"context"
+	"fmt"
+	"github.com/google/uuid"
+	"github.com/tidwall/gjson"
+	"strings"
+	"time"
+)
+
+type fhirTask struct {
+	data gjson.Result
+}
+
+func (task fhirTask) MarshalToTask() (*Task, error) {
+	if rType := task.data.Get("resourceType").String(); rType != "Task" {
+		return nil, fmt.Errorf("invalid resource type. got: %s, expected Task", rType)
+	}
+	codeQuery := fmt.Sprintf("code.coding.#(system==%s).code", SnomedCodingSystem)
+	if codeValue := task.data.Get(codeQuery).String(); codeValue != string(SnomedTransferCode) {
+		return nil, fmt.Errorf("unexpecting coding: %s", codeValue)
+	}
+	patientID := ""
+	if parts := strings.Split(task.data.Get("for.reference").String(), "/"); len(parts) > 1 {
+		patientID = parts[1]
+	}
+	return &Task{
+		ID: task.data.Get("id").String(),
+		TaskProperties: TaskProperties{
+			Status:      task.data.Get("status").String(),
+			OwnerID:     task.data.Get("owner.identifier.value").String(),
+			RequesterID: task.data.Get("requester.agent.identifier.value").String(),
+			PatientID:   patientID,
+		},
+	}, nil
```

Comment:
What's the reason for using `gjson` primitives instead of mapping plain Go structs (and parsing them from JSON)?

Reply:
This was initially because FHIR is so complex and requires a lots of Go structs and (custom) marshalling. I started to make structs for it anyways because I figured at least the marshalling part is actually not that hard. But unmarshalling is.

Maybe I'll take a look once more to see if I can find a library with a generated Go FHIR model.

Concerning this particular function: it is actually unused, I'll remove it.